src/tutorials/perks_tutorial_part2.md

```diff
@@ -0,0 +1,1609 @@
+
+                status.success = true
+                status.senderMessage = "Resource successfully given to " .. receiver.name .. "."
+                status.receiverMessage = "You have been gifted " .. tostring(commandData.resourceAmount) .. " " .. commandData.name .. "."
+            else
+                status.senderMessage = "Command data \"" .. item .. "\" does not exist."
+            end
+        end,
+
+        perk = function(receiver, item, status)
+            local commandData = CommandParser.GetCommandData(item)
+
+            if commandData ~= nil then
+                local playerData = Storage.GetPlayerData(receiver)
+
+                if commandData.perk ~= nil then
+                    if receiver:GetPerkTimeRemaining(commandData.perk) > 0 then
+                        status.senderMessage = "Player already has this time limited perk."
+                    else
+                        local durationSeconds = 15--commandData.duration * 60
+                        local expireSeconds = DateTime.CurrentTime().secondsSinceEpoch + durationSeconds
+                        local expireDateTime = DateTime.FromSecondsSinceEpoch(expireSeconds)
+
+                        playerData[commandData.storageKey] = expireDateTime.secondsSinceEpoch
+                        receiver:SetResource(commandData.resourceKey, 1)
+
+                        Storage.SetPlayerData(receiver, playerData)
+
+                        status.senderMessage = "Perk successfully given to " .. receiver.name .. "."
+                        status.receiverMessage = "You have been gifted " .. commandData.name .. "."
+                        status.success = true
+                    end
+                end
+            else
+                status.senderMessage = "Command data \"" .. item .. "\" does not exist."
+            end
+        end
+
+    })
+
+    CommandParser.AddCommand("remove", {
+
+        perk = function(receiver, item, status)
+            local commandData = CommandParser.GetCommandData(item)
+
+            if commandData ~= nil then
+                local playerData = Storage.GetPlayerData(receiver)
+
+                if commandData.perk ~= nil then
+                    if receiver:GetPerkTimeRemaining(commandData.perk) > 0 then
+                        status.senderMessage = "Player has purchased this perk."
+                    else
+                        playerData[commandData.storageKey] = 0
+                        receiver:SetResource(commandData.resourceKey, 0)
+
+                        Storage.SetPlayerData(receiver, playerData)
+
+                        status.senderMessage = "Perk successfully removed from " .. receiver.name .. "."
+                        status.success = true
+                    end
+                end
+            else
+                status.senderMessage = "Command data \"" .. item .. "\" does not exist."
+            end
+        end
+
+    })
+    ```
+
```

Comment:
perhaps a warning to update admins if they copy and pasted the whole file (like me)